### PR TITLE
SYB 016/plugins

### PR DESCRIPTION
**Pull request by vereis**
- Allow events to be extended by provided plugins, if any
- Bootstrap plugin behaviour
- Initial spike of an Absinthe plugin
- Initial spike of an Ecto plugin

image

image

### CODE REVIEW

File: lib/sibyl/plugins/absinthe.ex [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  This plugin will extend Sibyl and enable any configured Sibyl handler to listen to various events emitted
+  by `Absinthe` including field resolution, query execution, and batching.
+
+  Use via `Sibyl.Handlers.attach_all_events(plugins: [Sibyl.Plugins.Absinthe])`.
+  """
+
+  @behaviour Sibyl.Plugin
+  @behaviour Sibyl.Handler
+
+  @plugin_prefix [:sibyl, :plugins, :absinthe]
```
Comment (tomasz-tomczyk) [inline lib/sibyl/plugins/absinthe.ex:15]: I'd probably make it a callback on the behaviour too

File: lib/sibyl/plugins/absinthe.ex [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    [:absinthe, :execute, :operation, :start] => @plugin_prefix ++ [:operation, :start],
+    [:absinthe, :execute, :operation, :stop] => @plugin_prefix ++ [:operation, :stop],
+    [:absinthe, :resolve, :field, :start] => @plugin_prefix ++ [:resolve, :field, :start],
+    [:absinthe, :resolve, :field, :stop] => @plugin_prefix ++ [:resolve, :field, :stop],
+    [:absinthe, :middleware, :batch, :start] => @plugin_prefix ++ [:middleware, :batch, :start],
+    [:absinthe, :middleware, :batch, :stop] => @plugin_prefix ++ [:middleware, :batch, :stop]
+  }
+
+  @impl Sibyl.Plugin
+  def identity, do: Enum.join(@plugin_prefix, "-")
```
Comment (tomasz-tomczyk) [inline lib/sibyl/plugins/absinthe.ex:27]: Can we make that the default on the behaviour?

**Review by tomasz-tomczyk — APPROVED** [review]
❤️ 💛 🧡 💚 💙

File: lib/sibyl/plugins/absinthe.ex (hunk 1 above)
Comment (vereis) [reply lib/sibyl/plugins/absinthe.ex:15]: Yup good shout

**Pull request merged**